stellarphot/core.py

```diff
@@ -715,6 +800,225 @@ def catalog_name(self):
     def catalog_source(self):
         return self.meta['catalog_source']
 
+    @staticmethod
+    def _tidy_vizier_catalog(data, mag_column_regex, color_column_regex):
+        """
+        Transform a Vizier catalog with magnitudes into tidy structure. Or
+        at least tidier -- this only handles changing magnitude and color
+        columns into tidy format.
+
+        Parameters
+        ----------
+
+        data : `astropy.table.Table`
+            Table of catalog information. There are no restrictions on the columns.
+
+        mag_column_regex : str
+            Regular expression to match magnitude columns.
+
+        color_column_regex : str
+            Regular expression to match color columns.
+
+        Returns
+        -------
+
+        `astropy.table.Table`
+            Table with magnitude and color columns in tidy format. All other
+            columns are preserved as they were in the input data.
+        """
+
+        mag_re = re.compile(mag_column_regex)
+        color_re = re.compile(color_column_regex)
+
+        # Find all the magnitude and color columns
+        mag_match = [mag_re.match(col) for col in data.colnames]
+        color_match = [color_re.match(col) for col in data.colnames]
+
+        # create a single list of all the matches
+        matches = [m_match if m_match else c_match
+                   for m_match, c_match in zip(mag_match, color_match)]
+
+        # The passband should be the first group match.
+        passbands = [match[1] for match in matches if match]
```

Comment:
Are you grabbing the second character from the 'match' column name?  I *thought* it was usually the first character in the column name...  Also, if you include any color column names at all in matches, is passband a valid idea?

Reply:
The first match is the entire string (or at least the entire matched part of the string) and the next one is the first matching group, with the groups set up by the parentheses in the regex.

Reply:
Oh that's right, `re` is interesting that way...